### PR TITLE
[Test] Fix socket test on macos.

### PR DESCRIPTION
**Pull request by MediosZ**
# Overview
This PR fixes potential errors of sockets on macOS.
## 
`sizeof(sockaddr_in) - sizeof(sockaddr_in::sin_family);` on Linux is 14 but on macOS it's 15.
That's because sin_family on Linux is unsigned long, but __uint8_t on macOS.
Adding a condition to handle this on both platforms.

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review bot](https://github.com/flows-network/github-pr-summary/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

**Comment by MediosZ** [comment]
`setsockopt` doesn't pass the ci, but it works fine locally.
```txt
17: [ RUN      ] WasiSockTest.SockOpt
17: /Users/runner/work/WasmEdge/WasmEdge/test/host/socket/wasi_socket.cpp:1005: Failure
17: Value of: *MemInst.getPointer<const bool *>(ResBufPtr)
17:   Actual: false
17: Expected: true
17: [  FAILED  ] WasiSockTest.SockOpt (0 ms)
```

**Comment by MediosZ** [comment]
It turns out that we should not reinterpret_cast uint8_t* to bool*, because this is a undefined behavior.
[Reference](https://stackoverflow.com/questions/37418412/using-reinterpret-cast-with-bool)

**Comment by hydai** [comment]
@MediosZ Please rebase to the latest master branch. Thanks

File: test/host/socket/wasi_socket.cpp [hunk 1]
```diff
@@ -995,7 +995,9 @@ TEST(WasiSockTest, SockOpt) {
                        Errno);
 
     EXPECT_EQ(Errno[0].get<int32_t>(), __WASI_ERRNO_SUCCESS);
-    EXPECT_TRUE(*MemInst.getPointer<const int32_t *>(ResBufPtr));
+    // reinterpreting uint8_t* to bool* is a undefined behavior.
```
Comment (hydai) [inline test/host/socket/wasi_socket.cpp:998]: We can remove this line.

**Review by hydai — CHANGES_REQUESTED** [review]

File: test/host/socket/wasi_socket.cpp (hunk 1 above)
Comment (MediosZ) [reply test/host/socket/wasi_socket.cpp:998]: Got it

Comment (hydai) [reply test/host/socket/wasi_socket.cpp:997]: I mean the comment, not the `EXPECT_EQ(Errno[0].get<int32_t>(), __WASI_ERRNO_SUCCESS);` ;-)

**Review by hydai — CHANGES_REQUESTED** [review]

**Comment by MediosZ** [comment]
It is still needed, cuz it fixes another error on macOS and enable all tests in CI.

**Comment by hydai** [comment]
Ok, then let's merge it.

**Review by hydai — APPROVED** [review]

**Comment by hydai** [comment]
Thanks! @MediosZ

**Pull request merged**